VMAccess/vmaccess.py

```diff
@@ -382,10 +382,12 @@ def _allow_password_auth():
     _set_sshd_config(config, "PasswordAuthentication", "yes")
     ext_utils.replace_file_with_contents_atomic(SshdConfigPath, "\n".join(config))
 
-    if isinstance(MyDistro, dist_utils.UbuntuDistro): #handle ubuntu 22.04 (sshd_config.d directory)
+    if isinstance(MyDistro, dist_utils.UbuntuDistro) or \ #handle ubuntu 22.04 (sshd_config.d directory)
```

Comment:
I would suggest doing a sshd -T and verifies the output to check that Password authentication is indeed enabled after the restart of ssh. If the password authentication is still set to no, an error should be returned back to the user. 
This is the only way to ensure that the change did take effect as intended as there might be other .conf files that affect sshd behavior, not just 50-cloud-init.conf.

Reply:
I do not know how to do this.